base/src/main/java/io/spine/code/java/ClassNameNotation.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.code.java;
+
+import io.spine.annotation.Internal;
+
+/**
+ * Tools working with the Java class names.
+ *
+ * <p>A Java class name may be represented with several types: {@code ClassName},
+ * {@code NestedClassName}, {@code SimpleClassName}, etc. This utility handles the common string
+ * processing problems for those types.
+ */
+@Internal
+public final class ClassNameNotation {
```

Comment:
This class is used only by `ClassName` and `NestedClassName`.

`DOT_SEPARATOR` is used in many places, but in some of them the context is not a file name, but a package name. E.g. `Directory.of(PackagName)`.

I suggest to have the only method of this class made a part of `ClassName`. It would eliminate the utility class.